plugin/externalPluginEntry.go

```diff
@@ -390,6 +391,30 @@ func (e *externalPluginEntry) blockRead(ctx context.Context, size int64, offset
 	return inv.stdout.Bytes(), nil
 }
 
+func (e *externalPluginEntry) Write(ctx context.Context, p []byte) error {
```

Comment:
We could DRY this up by re-factoring invocation to something like
```
type invocation struct {
  internal.Command // So we get the "SetStdout", "SetStderr", "SetStdin" stuff for free
  // All the other stuff
}

// Add an InvokeAndWait helper to invocation and modify pluginScript's InvokeAndWait to call that.
// May also need to store context in invocation to record to activity?
```

Then in Write, this would read as
```
inv := e.script.NewInvocation(ctx, "write", e).SetStdout(...).SetStderr(...).SetStdin(...)
err := inv.InvokeAndWait()
etc.
```

Reply:
Maybe we could move over `SetStdout`, `SetStderr` to invocation#InvokeAndWait. Idea is `invocation#InvokeAndWait` would fill invocation#stdout, invocation#stderr.

Reply:
I can work on refactoring things a bit.